rust/src/interface.rs

```diff
@@ -0,0 +1,324 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! API traits for ADBC structs
+//!
+//! These are the interfaces to ADBC structs made more ergonomic for Rust
+//! developers. They are implemented by the structs in [crate::driver_manager].
+use arrow::{datatypes::Schema, record_batch::RecordBatch, record_batch::RecordBatchReader};
+
+use crate::ffi::AdbcObjectDepth;
+
+/// Databases hold state shared by multiple connections. This typically means
+/// configuration and caches. For in-memory databases, it provides a place to
+/// hold ownership of the in-memory database.
+pub trait DatabaseApi {
```

Comment:
nit: is it typical to suffix traits with 'Api'?

Reply:
This was mostly to differentiate between the `AdbcDatabase` in the driver_manager module and this trait. But given we want to be less FFI-centric, I renamed the driver_manager `AdbcDatabase` -> `DriverDatabase` and renamed this one `AdbcDatabase`.